packages/signals/src/global/mod.rs

```diff
@@ -1,18 +1,54 @@
 use dioxus_core::prelude::{provide_root_context, try_consume_context};
-use std::{any::Any, cell::RefCell, collections::HashMap, rc::Rc};
+use std::{any::Any, cell::RefCell, collections::HashMap, panic::Location, rc::Rc};
+use uuid::Uuid;
 
 mod memo;
 pub use memo::*;
 
 mod signal;
 pub use signal::*;
 
+use crate::Signal;
+
+pub(crate) static UUID_NAMESPACE: uuid::Uuid = uuid::Uuid::NAMESPACE_URL;
+
+/// The context for global signals
 #[derive(Clone)]
-pub(crate) struct GlobalSignalContext {
-    signal: Rc<RefCell<HashMap<*const (), Box<dyn Any>>>>,
+pub struct GlobalSignalContext {
+    signal: Rc<RefCell<HashMap<GlobalKey, Box<dyn Any>>>>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum GlobalKey {
```

Comment:
This doesn't work because you need to allocate the buffer somewhere at compile time. You could do it with a macro, but we don't want the entry point for global signals to be a macro

```rust
const BUFFER_SIZE: usize = 2usize.pow(16);
const fn location_to_str(location: &'static Location<'static>) -> [u8; BUFFER_SIZE] {
    let mut buffer = [0u8; BUFFER_SIZE];
    let mut fill_index = 0;
    // The first part of the location is the file name
    let file_name = location.file().as_bytes();
    while fill_index < file_name.len() {
        buffer[fill_index] = file_name[fill_index];
        fill_index += 1;
    }

    // Then the colon
    buffer[fill_index] = b':';
    fill_index += 1;
    // Then the line number
    let mut line_number = location.line();
    while line_number != 0 {
        let digit = line_number % 10;
        buffer[fill_index] = b'0' + digit as u8;
        fill_index += 1;
        line_number /= 10;
    }

    // Then the colon
    buffer[fill_index] = b':';
    fill_index += 1;
    // Then the file name
    let file_name = location.file().as_bytes();
    while fill_index < file_name.len() {
        buffer[fill_index] = file_name[fill_index];
        fill_index += 1;
    }

    let (filled, _) = buffer.split_at(fill_index);

    match std::str::from_utf8(filled) {
        Ok(s) => s,
        Err(_) => panic!("formatting the location created invalid utf-8"),
    }
}
```